### PR TITLE
Add blog post for type constraints (domains)

### PR DESCRIPTION
**Pull request by olirice**
## What kind of change does this PR introduce?
Adds a blog post for domain types in Postgres

### CODE REVIEW

File: apps/www/_blog/2023-01-25-type-constraints-in-50-lines-of-sql.mdx [hunk 1]
```diff
@@ -0,0 +1,271 @@
+---
+title: Type Constraints in 50 lines of SQL
+description: Creating validated data types in Postgres
+author: oli_rice
+image:
+thumb:
```
Comment (olirice) [inline apps/www/_blog/2023-01-25-type-constraints-in-50-lines-of-sql.mdx:6]: TODO: image here

File: apps/www/_blog/2023-01-25-type-constraints-in-50-lines-of-sql.mdx [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+To illustrate, we’ll create an `semver` data type to represent [Semantic Versioning](https://semver.org) values. We’ll then add validation rules to make invalid states unrepresentable.
+
+## SemVer
+
+A (very) loose primer on SemVer:
+
+SemVer is a specification for representing software versions that communicate information about backwards compatibility. The type is typically represented as a string with 5 components.
+
+<IMAGE HERE>
```
Comment (olirice) [inline apps/www/_blog/2023-01-25-type-constraints-in-50-lines-of-sql.mdx:31]: TODO: image here that is comparable to
Screenshot 2023-01-25 at 8 34 35 AM

**Review by w3b6x9 — APPROVED** [review]

File: apps/www/_blog/2023-01-25-type-constraints-in-65-lines-of-sql.mdx [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+
+To illustrate, we’ll create an `semver` data type to represent [Semantic Versioning](https://semver.org) values. We’ll then add validation rules to make invalid states unrepresentable.
+
+## SemVer
+
+A (very) loose primer on SemVer:
+
+SemVer is a specification for representing software versions that communicate information about backwards compatibility. The type is typically represented as a string with 5 components.
+
+![image](/images/blog/2023-jan/semver-spec.png)
```
Comment (alaister) [inline apps/www/_blog/2023-01-25-type-constraints-in-65-lines-of-sql.mdx:31]: I think this is a `.jpg`, not a `.png`

**Review by alaister — CHANGES_REQUESTED** [review]

File: apps/www/_blog/2023-01-25-type-constraints-in-65-lines-of-sql.mdx [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+```sql
+select
+    *
+from
+    package_version
+/*
+id | package_name |    package_semver
+-------------------------------------
+ 1 | supabase-js, |         (2,2,3,,)
+ 1 | supabase-js, | (2,0,0,"{rc,1}",)
```
Comment (alaister) [inline apps/www/_blog/2023-01-25-type-constraints-in-65-lines-of-sql.mdx:209]: Is the `,` wanted in `supabase-js,`?

File: apps/www/_blog/2023-01-25-type-constraints-in-65-lines-of-sql.mdx [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+    id,
+    package_name,
+    semver_to_text(package_semver) as ver -- cast as text
+from
+    package_version
+/*
+id | package_name |   ver
+------------------------------
+ 1 | supabase-js, |      2.2.3
+ 1 | supabase-js, | 2.0.0-rc.1
```
Comment (alaister) [inline apps/www/_blog/2023-01-25-type-constraints-in-65-lines-of-sql.mdx:251]: More `,`

**Review by alaister — DISMISSED** [review]

**Pull request merged**